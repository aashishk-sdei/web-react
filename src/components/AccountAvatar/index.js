import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Avatar from "@material-ui/core/Avatar";
import { grey } from "@material-ui/core/colors";

const colors = ["#5199B5", "#388367", "#885F90", "#B88B82", "#8A4831", "#BC8A40"];
const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    width: 32,
    height: 32,
    "& > *": {
      margin: 0,
    },
  },
  small: {
    width: theme.spacing(4),
    height: theme.spacing(4),
  },
  grey: (props) => {
    return {
      color: theme.palette.getContrastText(grey[900]),
      backgroundColor: props.bgColorCode || props.bgColorCode === 0 ? colors[props.bgColorCode] : grey[900],
      width: theme.spacing(4),
      height: theme.spacing(4),
      fontSize: props.likedPersons && "12px",
    };
  },
}));

const AccountAvatar = ({ avatarName, imgSrc, ...props }) => {
  const classes = useStyles(props);

  return <div className={props.avatarsizeclass?props.avatarsizeclass:"avtar-circle-medium"}>{imgSrc ? <Avatar src={imgSrc} className={classes.small} /> : <Avatar className={classes.grey}>{avatarName.toUpperCase()}</Avatar>}</div>;
};

export default AccountAvatar;
