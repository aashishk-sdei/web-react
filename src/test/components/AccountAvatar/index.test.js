import React from 'react';
import { shallow } from 'enzyme';
import AccountAvatar from '../../../components/AccountAvatar';

test('should test AccountAvatar component with list of props', () => {
    const wrapper = shallow(
        <AccountAvatar avatarName={'UserName'} imgSrc={undefined} />
    );
    expect(wrapper).toMatchSnapshot();
});


test('should test div type in AccountAvatar component', () => {
    const wrapper = shallow(<AccountAvatar avatarName={'UserName'} imgSrc={undefined}/>);
    const input = wrapper.find('div');
    expect(input).toHaveLength(1);
});