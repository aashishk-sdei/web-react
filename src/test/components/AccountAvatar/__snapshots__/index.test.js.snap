// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test AccountAvatar component with list of props 1`] = `ShallowWrapper {}`;
